app/src/org/commcare/activities/connect/ConnectActivity.java

```diff
@@ -147,4 +232,47 @@ public void startAppValidation() {
         i.putExtra(CommCareVerificationActivity.KEY_LAUNCH_FROM_SETTINGS, true);
         verificationLauncher.launch(i);
     }
+
+    public void getJobDetails() {
+        ApiConnect.getConnectOpportunities(ConnectActivity.this, new IApiCallback() {
+            @Override
+            public void processSuccess(int responseCode, InputStream responseData) {
+                try {
+                    String responseAsString = new String(StreamsUtil.inputStreamToByteArray(responseData));
+                    if (responseAsString.length() > 0) {
+                        //Parse the JSON
+                        JSONArray json = new JSONArray(responseAsString);
+                        List<ConnectJobRecord> jobs = new ArrayList<>(json.length());
+                        for (int i = 0; i < json.length(); i++) {
+                            JSONObject obj = (JSONObject) json.get(i);
+                            ConnectJobRecord job = ConnectJobRecord.fromJson(obj);
+                            jobs.add(job);
+                            if (job.getJobId() == Integer.parseInt(opportunityId)) {
+                                ConnectManager.setActiveJob(job);
+                            }
+                        }
+
+                        setFragmentRedirection();
+                    }
+                } catch (IOException | JSONException | ParseException e) {
+                    Logger.exception("Parsing return from Opportunities request", e);
+                }
+            }
+
+            @Override
+            public void processFailure(int responseCode, IOException e) {
+                Logger.log("ERROR", String.format(Locale.getDefault(), "Opportunities call failed: %d", responseCode));
```

Comment:
What does the user see when this happens? Thinking they should be taken to the Connect jobs list and somehow shown a message saying there was a problem talking to the server

Reply:
I set a flag to check if an API call is successful. If the flag indicates success, redirection to targeted fragment occurs; otherwise, it does not. 

Additionally, I display a toast message to indicate errors.

Let me know if you would like any changes.

Reply:
I'll need to test this out to see what the user experience is like. Okay to leave for now.

Reply:
Okay